### PR TITLE
[KHWEBA-67] Add account 

### PR DESCRIPTION
**Pull request by OlgaOrlova**
image
image

### CODE REVIEW

File: src/app/accounts/accounts/accounts.component.ts [hunk 1]
```diff
@@ -1,37 +1,31 @@
-import { Component, OnInit } from "@angular/core";
+import { AfterContentInit, Component, OnInit } from "@angular/core";
 import { MatDialog } from "@angular/material";
 
 import { AccountDialogComponent } from "../account-dialog/account-dialog.component";
 import { Account } from "../../interfaces/account";
+import { DatabaseService } from "../../services/database.service";
```
Comment (gift-a) [inline src/app/accounts/accounts/accounts.component.ts]: As I understood, you need to make your own service as we've done in vanilla

File: src/app/accounts/accounts/accounts.component.ts [hunk 2]
```diff
@@ -44,7 +38,15 @@ export class AccountsComponent implements OnInit {
     });
     addAccountDialog
       .afterClosed()
-      .subscribe(res => (res ? console.log(res) : null));
+      .flatMap(res => {
+        return this.database.setData("accounts", res);
+      })
+      .flatMap(() => {
+        return this.database.getList("accounts");
```
Comment (gift-a) [inline src/app/accounts/accounts/accounts.component.ts]: do you need to get data from server twice? Observable will rerender view on every change (on row 21). or what is the purpose of this `getlist`?

File: src/app/accounts/accounts/accounts.component.ts [hunk 3]
```diff
@@ -44,7 +38,15 @@ export class AccountsComponent implements OnInit {
...
       .afterClosed()
-      .subscribe(res => (res ? console.log(res) : null));
+      .flatMap(res => {
+        return this.database.set(res);
+      })
+      .flatMap(() => {
+        return this.database.getAll();
+      })
+      .subscribe(result => {
+        this.accounts = (result as Account[]).reverse();
```
Comment (ihor7016) [inline src/app/accounts/accounts/accounts.component.ts]: accounts service should return data type `Account`

File: src/app/accounts/accounts/accounts.component.ts [hunk 4]
```diff
@@ -1,37 +1,31 @@
-import { Component, OnInit } from "@angular/core";
+import { AfterContentInit, Component, OnInit } from "@angular/core";
```
Comment (gift-a) [inline src/app/accounts/accounts/accounts.component.ts]: if you don't use onInit, just remove it

File: src/app/accounts/accounts/accounts.component.ts [hunk 5]
```diff
@@ -1,37 +1,31 @@
...
-  ngOnInit() {
-    this.accounts = [
-      {
-        id: "id1",
-        name: "Privat",
-        balance: 500,
-        type: "savings",
-        currency: "€"
+  ngAfterContentInit() {
+    this.database.getList().subscribe(
```
Comment (gift-a) [inline src/app/accounts/accounts/accounts.component.ts]: why do you get list in `afterContentInit`, but not in `onInit`?

File: src/app/accounts/accounts/accounts.component.ts [hunk 6]
```diff
@@ -1,37 +1,31 @@
...
       },
-      { id: "id2", name: "Cash", balance: 2000, type: "cash", currency: "₴" },
-      {
-        id: "id3",
-        name: "BoaBank",
-        balance: 5000,
-        type: "credit",
-        currency: "$"
+      error => {
+        console.log(error);
```
Comment (ihor7016) [inline src/app/accounts/accounts/accounts.component.ts]: please remove error handler  if you just need to console.log

File: src/app/services/abstract.service.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { DatabaseService } from "./database.service";
+import { Observable } from "rxjs/Observable";
+export abstract class AbstractService<T> {
```
Comment (ihor7016) [inline src/app/services/abstract.service.ts]: please separate imports and declarations with newline

**Review by gift-a — APPROVED** [review]

**Pull request merged**